terraform/deployment.tf

```diff
@@ -0,0 +1,144 @@
+resource "kubernetes_deployment" "pomerium" {
+  metadata {
+    name      = var.deployment_name
+    namespace = var.namespace_name
+    labels    = var.deployment_labels
+  }
+
+  lifecycle {
+    ignore_changes = [
+      metadata[0].annotations
+    ]
+  }
+
+  spec {
+    replicas = var.deployment_replicas
+
+    selector {
+      match_labels = {
+        "app.kubernetes.io/name" = "pomerium-ingress-controller"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          "app.kubernetes.io/name" = "pomerium-ingress-controller"
+        }
+      }
+
+      spec {
+        service_account_name             = kubernetes_service_account.controller.metadata[0].name
+        termination_grace_period_seconds = 10
+
+        security_context {
+          run_as_non_root = true
+        }
+
+        node_selector = merge(local.default_node_selector, var.node_selector)
```

Comment:
optional: It's sometimes nicer to do a merge in the local variable.

Reply:
do you mean : 

```terraform
node_selector = local.node_selector
```

and then have 
```terraform
locals {
  default_node_selector = {...}
  node_selector = merge(local.default_node_selector, var.node_selector)
}
```

Reply:
Yes,  Or `local.computed_node_selector` to denote it's not a single-sourced value.